addon/-private/system/store/finders.js

```diff
@@ -143,14 +143,14 @@ export function _query(adapter, store, modelName, query, recordArray) {
 
   if (createRecordArray) {
     recordArray = recordArray || store.recordArrayManager.createAdapterPopulatedRecordArray(modelName, query);
-    promise = Promise.resolve().then(() => adapter.query(store, modelClass, query, recordArray));
+    promise = Promise.resolve().then(() => adapter.query(store, modelClass, query, recordArray, options));
   } else {
     promise = Promise.resolve().then(() => adapter.query(store, modelClass, query));
```

Comment:
Was it intentional that `options` isn't passed along if `createRecordArray` is false? There isn't such a distinction in [the documentation](https://api.emberjs.com/ember-data/3.10/classes/DS.Store/methods/createRecord?anchor=query) about when an included `adapterOptions` hash will be passed to `adapter.query`.

Reply:
I think it was a mistake actually.

@runspired / @hjdivad / @igorT can you sanity check that?

Reply:
Speaking with @runspired he reminded me what this was intentionally, as without a record array, we cannot pass `options` as a fourth argument without breaking user-land arity checks. Also, creating the record array eagerly and providing it here, had some performance implications.

The plan at the time, was to eventually change this API to use an object, instead of positional arguments, to avoid this problem.

Reply:
Additional note: if the arity check says that `adapter.query` uses less than 4 arguments (recordArray is 4th, options is 5th), we don't pass either along.

I'm not sure we were as worried about breaking user-land arity checks as we were about the performance costs of materializing the snapshotRecordArray when we didn't need to.

If your `query` implementation has an arity of 4/5 and is still not receiving options, then let's dive into that; however, simply adding it into the function signature in prep for usage should be enough to receive any options that were passed in.

Reply:
@runspired and I dug into this more, and there's an issue with the arity check where it doesn't properly detect the argument length for adapter.query. This is captured in #6232.